crates/db_views/src/comment_report_view.rs

```diff
@@ -117,456 +111,7 @@ fn queries<'a>() -> Queries<
-  pub async fn read(
-    pool: &mut DbPool<'_>,
-    report_id: CommentReportId,
-    my_person_id: PersonId,
-  ) -> Result<Self, Error> {
-    queries().read(pool, (report_id, my_person_id)).await
-  }
-
-  /// Returns the current unresolved comment report count for the communities you mod
-  pub async fn get_report_count(
-    pool: &mut DbPool<'_>,
-    my_person_id: PersonId,
-    admin: bool,
-    community_id: Option<CommunityId>,
-  ) -> Result<i64, Error> {
-    use diesel::dsl::count;
-
-    let conn = &mut get_conn(pool).await?;
-
-    let mut query = comment_report::table
-      .inner_join(comment::table)
-      .inner_join(post::table.on(comment::post_id.eq(post::id)))
-      .filter(comment_report::resolved.eq(false))
-      .into_boxed();
-
-    if let Some(community_id) = community_id {
-      query = query.filter(post::community_id.eq(community_id))
-    }
-
-    // If its not an admin, get only the ones you mod
-    if !admin {
-      query
-        .inner_join(
-          community_actions::table.on(
-            community_actions::community_id
-              .eq(post::community_id)
-              .and(community_actions::person_id.eq(my_person_id))
-              .and(community_actions::became_moderator.is_not_null()),
-          ),
-        )
-        .select(count(comment_report::id))
-        .first::<i64>(conn)
-        .await
-    } else {
-      query
-        .select(count(comment_report::id))
-        .first::<i64>(conn)
-        .await
-    }
-  }
-}
-
-#[derive(Default)]
-pub struct CommentReportQuery {
-  pub community_id: Option<CommunityId>,
-  pub comment_id: Option<CommentId>,
-  pub page: Option<i64>,
-  pub limit: Option<i64>,
-  pub unresolved_only: bool,
-}
-
-impl CommentReportQuery {
-  pub async fn list(
-    self,
-    pool: &mut DbPool<'_>,
-    user: &LocalUserView,
-  ) -> Result<Vec<CommentReportView>, Error> {
-    queries().list(pool, (self, user)).await
-  }
-}
-
-#[cfg(test)]
-#[expect(clippy::indexing_slicing)]
-mod tests {
-
-  use crate::{
-    comment_report_view::{CommentReportQuery, CommentReportView},
-    structs::LocalUserView,
-  };
-  use lemmy_db_schema::{
-    aggregates::structs::CommentAggregates,
-    source::{
-      comment::{Comment, CommentInsertForm},
-      comment_report::{CommentReport, CommentReportForm},
-      community::{Community, CommunityInsertForm, CommunityModerator, CommunityModeratorForm},
-      instance::Instance,
-      local_user::{LocalUser, LocalUserInsertForm},
-      local_user_vote_display_mode::LocalUserVoteDisplayMode,
-      person::{Person, PersonInsertForm},
-      post::{Post, PostInsertForm},
-    },
-    traits::{Crud, Joinable, Reportable},
-    utils::{build_db_pool_for_tests, RANK_DEFAULT},
-    CommunityVisibility,
-    SubscribedType,
-  };
-  use lemmy_utils::error::LemmyResult;
-  use pretty_assertions::assert_eq;
-  use serial_test::serial;
-
-  #[tokio::test]
-  #[serial]
-  async fn test_crud() -> LemmyResult<()> {
```

Comment:
I simplified and moved all the tests to `ReportCombinedView`, since they need to call `ReportCombinedView::list`